src/implementation/Server/DaprServer.ts

```diff
@@ -50,43 +45,36 @@ export default class DaprServer {
   readonly actor: IServerActor;
   readonly client: DaprClient;
 
-  constructor(
-    serverHost?: string,
-    serverPort?: string,
-    daprHost?: string,
-    daprPort?: string,
-    communicationProtocol: CommunicationProtocolEnum = CommunicationProtocolEnum.HTTP,
-    clientOptions: Partial<DaprClientOptions> = {},
-    serverOptions: DaprServerOptions = {},
-  ) {
-    this.serverHost = serverHost ?? Settings.getDefaultHost();
-    this.serverPort = serverPort ?? Settings.getDefaultAppPort(communicationProtocol);
-    this.serverOptions = serverOptions;
-    this.daprHost = daprHost ?? Settings.getDefaultHost();
-    this.daprPort = daprPort ?? Settings.getDefaultPort(communicationProtocol);
-
-    clientOptions.daprHost = this.daprHost;
-    clientOptions.daprPort = this.daprPort;
-    clientOptions.communicationProtocol = communicationProtocol;
-
+  constructor(serverOptions: Partial<DaprServerOptions> = {}) {
+    const communicationProtocol = serverOptions.communicationProtocol ?? Settings.getDefaultCommunicationProtocol();
+    const clientOptions = getClientOptions(serverOptions.clientOptions, communicationProtocol);
```

Comment:
Since we are reusing communication protocol, should we be consistent and also reuse logger options?